core/src/main/java/jenkins/model/lazy/AbstractLazyLoadRunMap.java

```diff
@@ -460,7 +473,8 @@ protected R load(int n, Index editInPlace) {
      *      If non-null, update this data structure.
      *      Otherwise do a copy-on-write of {@link #index}
      */
-    protected synchronized R load(File dataDir, Index editInPlace) {
+    private R load(File dataDir, Index editInPlace) {
+        assert Thread.holdsLock(this);
```

Comment:
Well, I made it `private` because I could. But yes that also makes it easier to demonstrate that callers hold the lock.